insights/parsers/softnet_stat.py

```diff
@@ -0,0 +1,100 @@
+"""
+softnet_stat - /proc/net/softnet_stat
+=====================================
+
+This parser parses the stats from network devices. These stats includes events
+per cpu(in row), number of packets processed i.e packet_process (first column),
+number of packet drops packet_drops (second column), time squeeze i.e net_rx_action
+performed time_squeeze(third column),cpu collision i.e collision occur while obtaining
+device lock while transmitting cpu_collision packets (Ninth column), received_rps
+number of times cpu woken up received_rps (tenth column), number of times reached flow
+limit count flow_limit_count (eleventh column).
+
+    The typical contents of ``cat /proc/net/softnet_stat`` file is as below::
+
+        00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+
+    Examples:
+        >>> SOFTNET_STATS = '''
+        ... 00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ...'''.strip()
+        >>> softnet_obj = shared[SoftNetStats]
```

Comment:
Declaring SOFTNET_STATS in the code usually implies that you're going to initialise the parser object yourself, e.g. by `softnet_obj = SoftNetStats(context_wrap(SOFTNET_STATS))`.  If you're pulling the declaration out of the shared data, then you don't need to declare SOFTNET_STATS.  I know that there's both styles in use in the code; I'd recommend using the `context_wrap` method as the examples are written as if you're in an interactive Python shell.

Reply:
Sure thing